raven/utilities/mk_test.py

```diff
@@ -0,0 +1,202 @@
+    (satkumartomer@gmail.com) which can be found at:
+    http://vsp.pnnl.gov/help/Vsample/Design_Trend_Mann_Kendall.htm
+    As stated on the webpage in the URL above the method uses a Monte-Carlo
+    simulation to determine the required number of points in time, n, to take a
+    measurement in order to detect a linear trend for specified small
+    probabilities that the MK test will make decision errors. If a non-linear
+    trend is actually present, then the value of n computed by VSP is only an
+    approximation to the correct n. If non-detects are expected in the
+    resulting data, then the value of n computed by VSP is only an
+    approximation to the correct n, and this approximation will tend to be less
+    accurate as the number of non-detects increases.
+    Input:
+        beta: probability of falsely accepting the null hypothesis
+        delta: change per sample period, i.e., the change that occurs between
+               two adjacent sampling times
+        std_dev: standard deviation of the sample points.
+        alpha: significance level (0.05 default)
+        n: initial number of sample points (4 default).
+        num_iter: number of iterations of the Monte-Carlo simulation (1000
+                  default).
+        tol: tolerance level to decide if the predicted probability is close
+             enough to the required statistical power value (1e-6 default).
+        num_cycles: Total number of cycles of the simulation. This is to ensure
+                    that the simulation does finish regardless of convergence
+                    or not (10000 default).
+        m: if the tolerance is too small then the simulation could continue to
+           cycle through the same sample numbers over and over. This parameter
+           determines how many cycles to look back. If the same number of
+           samples was been determined m cycles ago then the simulation will
+           stop.
+        Examples
+        --------
+          >>> num_samples = check_num_samples(0.2, 1, 0.1)
+    """
+    # Initialize the parameters
+    power = 1.0 - beta
+    P_d = 0.0
+    cycle_num = 0
+    min_diff_P_d_and_power = abs(P_d - power)
+    best_P_d = P_d
+    max_n = n
+    min_n = n
+    max_n_cycle = 1
+    min_n_cycle = 1
+    # Print information for user
+    print("Delta (gradient): {}".format(delta))
+    print("Standard deviation: {}".format(std_dev))
+    print("Statistical power: {}".format(power))
+
+    # Compute an estimate of probability of detecting a trend if the estimate
+    # Is not close enough to the specified statistical power value or if the
+    # number of iterations exceeds the number of defined cycles.
+    while abs(P_d - power) > tol and cycle_num < num_cycles:
+        cycle_num += 1
+        print("Cycle Number: {}".format(cycle_num))
+        count_of_trend_detections = 0
+
+        # Perform MK test for random sample.
+        for i in xrange(num_iter):
```

Comment:
`xrange` doesn't exist in python3. This will fail.

Reply:
OK code should be python3 friendly now.